lib/dataset/test/bin_test.cpp

```diff
@@ -198,6 +198,21 @@ TEST_P(BinTest, rebin_using_attr) {
   EXPECT_EQ(expected, result);
 }
 
+TEST_P(BinTest, rebin_using_attr_in_new_dimension) {
+  auto table = GetParam();
+  auto expected_temp = bin(table, {edges_x});
+  const auto z_coord = makeVariable<double>(Dims{Dim::X}, Shape{4},
+                                            Values{-10., -5.0, 0.5, 7.5});
+  expected_temp.coords().set(Dim::Z, z_coord);
+  const auto edges_z_coarse =
```

Comment:
Is it relevant that the key is not `Dim::X` (matching the dim label)?

Reply:
It needs to be a key that is not found in the original data that was binned.
That data has `Dim::X` and `Dim::Y` when it is created.
I need to have a coordinate that has `X` as a dimension, but is only defined on the wrapper level.